it/google-cloud-platform/src/main/java/org/apache/beam/it/gcp/bigquery/conditions/BigQueryRowsCheck.java

```diff
@@ -77,7 +90,15 @@ public static Builder builder(BigQueryResourceManager resourceManager, TableId t
   }
 
   public Long getRowCount() {
-    return resourceManager().getRowCount(tableId().getTable());
+    try {
+      return resourceManager().getRowCount(tableId().getTable());
+    } catch (Exception e) {
+      if (ExceptionUtils.containsMessage(e, "Not found: Table")) {
```

Comment:
It seems kind of odd to handle this exception at this level, especially since there's only one caller who does conditional logic on top anyways. Could we more cleanly just handle this in `check` instead?

Reply:
Are you saying like this?

Reply:
Yeah, I think that is much cleaner - thanks